### PR TITLE
Adding Ubuntu builds to ADO pipeline

### PR DESCRIPTION
**Pull request by dsgouda**

### CODE REVIEW

File: .azure-pipelines/mgmt.yml [hunk 1]
```diff
@@ -1,33 +1,46 @@
...
-      testRunTitle: 'Windows Test Results'
-      testResultsFormat: 'VSTest'
-      mergeTestResults: true
\ No newline at end of file
+    steps:
+        - task: DotNetCoreInstaller@0
+          displayName: 'Use .NET Core sdk $(DotNetCoreVersion)'
+          inputs:
+            version: '$(DotNetCoreVersion)'
+        - script: './tools/linuxScripts/runNetCoreSdkTests.sh'
```
Comment (weshaggard) [inline .azure-pipelines/mgmt.yml]: I realize this is a direct translation from your travis job but I would suggest eliminating this sh script and putting the build commands in the yml.

File: .azure-pipelines/mgmt.yml [hunk 2]
```diff
@@ -1,33 +1,46 @@
...
+          displayName: 'Build and test on Non-windows machine'
```
Comment (weshaggard) [inline .azure-pipelines/mgmt.yml]: Are you running at tests as part of that script? If so you should publish the test results here.

File: .azure-pipelines/mgmt.yml (hunk 1 above)
Comment (dsgouda) [reply .azure-pipelines/mgmt.yml]: @shahabhijeet to make the call here
From what I see this script actually picks exactly which projects to build and test which translating into azure pipeline is a fair bit of work, given we might be able to use the cross plat build tools which can simply build all the repo as is I wasn't sure if we wanted to port the script to azure pipeline steps, @shahabhijeet thoughts?

**Comment by weshaggard** [comment]
I assume you are looking at the failures in your new steps? https://dev.azure.com/azure-sdk/public/_build/results?buildId=22228

**Comment by dsgouda** [comment]
@weshaggard We are running into those errors in the current Travis builds. @shahabhijeet is working on fixing them.

**Comment by dsgouda** [comment]
@weshaggard please approve

File: .azure-pipelines/mgmt.yml [hunk 3]
```diff
@@ -5,31 +5,54 @@ pr:
...
+      vmImage: 'ubuntu-16.04'
+  
+    steps:
+        - task: DotNetCoreInstaller@0
+          displayName: 'Use .NET Core sdk $(DotNetCoreVersion)'
+          inputs:
+            version: '$(DotNetCoreVersion)'
+        - script: './tools/linuxScripts/runNetCoreSdkTests.sh'
+          displayName: 'Build and test on Non-windows machine'
+        - task: PublishTestResults@2
```
Comment (weshaggard) [inline .azure-pipelines/mgmt.yml]: I don't see any Ubuntu Test Results in the CI job so I must assume there aren't any .trx files produced from the runNetCoreSdkTests.sh script. While this is't a blocker it is something that should be investigated if you guys stay using that script longer term.

File: .azure-pipelines/mgmt.yml [hunk 4]
```diff
@@ -5,31 +5,54 @@ pr:
...
+  DotNetCoreVersion: '2.1.503'
+
+jobs:
+  - job: Windows_Build
+    pool:
+      vmImage: 'VS2017-Win2016'
+    variables:
+      IgnoreBuildTokens: 'Microsoft.Rest.ClientRuntime.Tracing.Tests Microsoft.Azure.Services.AppAuthentication.IntegrationTests KeyVault\dataPlane LocationBasedServices Az.Auth.Tests BotService.Tests Search Batch.FileStaging.Tests Azure.Batch.IntegrationTests Azure.Batch.ProtocolTests Batch\Support\FileConventions AzureBatchFileConventions.IntegrationTests Batch\DataPlane\Azure.Batch.Unit.Tests Batch\DataPlane\Tools\ConfigureAwaitAnalyzer\ConfigureAwaitAnalyzer.Tests ContainerService\ContainerService.Tests Consumption\Consumption.Tests Billing\Billing.Tests ApplicationInsights\Management\ApplicationInsights.Tests Batch\Management\Management.Batch.Tests SecurityCenter\SecurityCenter.Tests AzureStack\Admin\FabricAdmin\Fabric.Admin.Tests IotHub\IotHub.Tests Monitor\Monitor.Tests Compute\Compute.Tests Reservations\Reservations.Tests NetApp\NetApp.Tests ContainerRegistry\ContainerRegistry.Tests OperationalInsights\Management\OperationalInsights.Test Subscription\Subscription.Tests Cdn\Cdn.Tests'  
+      FlpArg: 'Summary;Verbosity=minimal;LogFile=msbuild.sum'
```
Comment (weshaggard) [inline .azure-pipelines/mgmt.yml]: Nit: I would simply make this one logging option like `LoggingArgs: '/flp:Summary;Verbosity=minimal;LogFile=msbuild.sum /flp1:warningsonly;logfile=msbuild.wrn /flp2:errorsonly;logfile=msbuild.err`

**Review by weshaggard — APPROVED** [review]
I left a few comments but otherwise the job looks good.  However as we chatted we will want to try and merge with the client jobs once you guys get the engineering tools updated.

File: .azure-pipelines/mgmt.yml (hunk 3 above)
Comment (dsgouda) [reply .azure-pipelines/mgmt.yml]: @weshaggard we have disabled tests on Linux as of today but might enable it at some point, hence the absence of .trx files.

**Pull request merged**